eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Tx.hs

```diff
@@ -367,20 +367,29 @@ totExUnits tx =
 -- Figure 6:Indexing script and data objects
 -- ===============================================================
 
-data ScriptPurpose c
-  = Minting !(PolicyID c)
-  | Spending !(TxIn c)
-  | Rewarding !(RewardAcnt c) -- Not sure if this is the right type.
-  | Certifying !(DCert c)
-  deriving (Eq, Show, Generic, NoThunks, NFData)
-
-instance (Crypto c) => EncCBOR (ScriptPurpose c) where
-  encCBOR (Minting x) = encode (Sum Minting 0 !> To x)
-  encCBOR (Spending x) = encode (Sum Spending 1 !> To x)
-  encCBOR (Rewarding x) = encode (Sum Rewarding 2 !> To x)
+data ScriptPurpose era
+  = Minting !(PolicyID (EraCrypto era))
+  | Spending !(TxIn (EraCrypto era))
+  | Rewarding !(RewardAcnt (EraCrypto era)) -- Not sure if this is the right type.
```

Comment:
```suggestion
  | Rewarding !(RewardAcnt (EraCrypto era))
```
It's already in production, so I sure hope this turned out to be the right type

Reply:
:facepalm: yes, that type is correct. and that comment should have been removed. thank you!!